frontend/express/public/core/events/javascripts/countly.details.views.js

```diff
@@ -107,19 +300,48 @@ var allEventsVuex = [{
     clyModel: countlyAllEvents
 }];
 
+var compareEventsVuex = [{
+    clyModel: countlyCompareEvents
+}];
+
 var AllEventsViewWrapper = new countlyVue.views.BackboneWrapper({
-    component: AllEventsView,
+    component: eventsWrapper,
     vuex: allEventsVuex,
     templates: [
-        "/core/events/templates/allEvents.html"
+        "/core/events/templates/allEvents.html",
+        "/core/events/templates/compareEvents.html"
+    ]
+});
+
+var CompareEventsViewWrapper = new countlyVue.views.BackboneWrapper({
+    component: eventsWrapper,
+    vuex: compareEventsVuex,
+    templates: [
+        "/core/events/templates/allEvents.html",
+        "/core/events/templates/compareEvents.html"
     ]
 });
 
 app.route("/analytics/events/key/:eventKey", "all-events", function(eventKey) {
     var params = {
-        eventKey: eventKey
+        eventKey: eventKey && eventKey !== "undefined" ? eventKey : undefined
     };
+    AllEventsViewWrapper.params = params;
+    this.renderWhenReady(AllEventsViewWrapper);
+});
 
+app.route("/analytics/events", "all-events", function() {
+    var params = {
+        eventKey: undefined
+    };
     AllEventsViewWrapper.params = params;
     this.renderWhenReady(AllEventsViewWrapper);
+});
+
+app.route("/analytics/events/compare", "compare-events", function() {
```

Comment:
you missed crud validation here - https://github.com/Countly/countly-server/pull/1962/files#diff-0c95dc88ba74b8f804566618dd218d450558b3e2039033280b5dee9a6c51dc16R322

also please wrap the file within an anonymous function 
```
(function(){
//
})()
```